src/Components/Error/Error.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { useEffect } from "react";
+import classNames from "classnames";
+
+type Props = {
+  error: string;
+  onClear: () => void;
+};
+
+export const Error: React.FC<Props> = ({ error, onClear }) => {
+  useEffect(() => {
+    if (error.length > 0) {
+      setTimeout(onClear, 3000);
```

Comment:
You should clear the timeout on the component unmount 
https://bobbyhadz.com/blog/react-cleartimeout